### ISSUE TITLE
Add second approver

### ISSUE THREAD

**Issue by PatrickFist** [timeoff-management/application#82]
We have teams/departments with up to 4 leads.
It is only possible to add a single approver per department.

I recommend at least a second approver per department.

**Comment by vpp** [comment]
Hi @PatrickFist ,

true, it is feature is missing. There were some other people asking for similar thing.

I will move it to higher priority.

Thank you for reporting.

**Comment by enriquetoledano** [comment]
I also think this is necesary.
In my case, I would need that the second approver were member of other department.

Thanks!

**Comment by vpp** [comment]
Thanks @enriquetoledano, i will push the issue higher.

**Comment by PatrickFist** [comment]
Sounds good, let me clarify following:

I hope to get two layers/tiers of approval. Three people will be involved:

- Holiday Requestor
- Holiday Replacement
- Supervisor Approval

Example:
**Tier1:**
Holiday Requestor
Holiday Replacement (some collegue confirming to take care of Requestors Jobs in the holiday time).
**Tier2:**
Supervisor A
Supervisor B

One Supervisor would be enough to approve the holiday asset.
And Holiday Requestor can choose any user within the system to set him as replacement each time he requests holiday. This association should not be fixed to somebody.

Thats my recommendation, hope you can do it this way Pavlo.

Thank you.
Patrick.

**Comment by vpp** [comment]
hi @PatrickFist, great description! 

Such feedback is really appreciated. 

I am going to look at this task after Timezones (#58) and Carry over to next year features are done.

**Comment by AsimPervez93** [comment]
@vpp 

I know this is frequently asked as per the tag but I'd just like to express that I would find this very useful in my instance.

Kind Regards,

**Comment by vpp** [comment]
As per feedback:

Thinking again about this: different actions have different "approvers".

* Requesting absence: managers and manager deledates. Optionally auto approve.
* Canceling Approved absence: self; manager, delegated managers. Optionally auto approve.
* Canceling Pending Absence : self; manager, delegated managers. Optionally auto approve.

**Comment by vpp** [comment]
Consider adding all "admins" as approvers.

**Comment by theapprover** [comment]
In my instance there is interest in having for the second tier of approval in the workflow that all go leaves approved by all of the Managers then go to a final person for ultimate approval.

e.g.
Requester A submits Vacation Form
Manager is sent email
Manager Approves the Vacation Request
Resources is sent an email
Resources does final Approval of Vacation Request

If at any stage it is rejected then the Requester is informed of the rejection

**Comment by vpp** [comment]
The functionality of secondary approvers was released into master.

Now it is possible to set secondary approvers per each department who will be able to act on requests coming from related users.

There are rough edges still like not all emails are sent to secondary supervisors, but I decided to release the feature to get early feedback.

Any other work to address issues corresponding to this feature are going to be done as separate ticket.

There are some very useful ideas in comments above, I will mark them so future development is aware of it, but I am not considering implementing them in scope of current ticket.

I am closing this ticket.
